components/src/dynamo/common/utils/graceful_shutdown.py

```
@@ -0,0 +1,114 @@
+    shutdown_event: Optional[asyncio.Event] = None,
+    grace_period_s: Optional[float] = None,
+) -> None:
+    def signal_handler() -> None:
+        asyncio.create_task(
+            graceful_shutdown_with_discovery(
+                runtime,
+                endpoints,
+                shutdown_event=shutdown_event,
+                grace_period_s=grace_period_s,
+            )
+        )
```

Ruff (0.14.14)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)